pkg/controller/common/util_test.go

```diff
@@ -234,6 +235,69 @@ var _ = Describe("Rebind", func() {
 		//make sure annotations of pv from before rebind dont get deleted
 		Expect(pv.Annotations["someAnno"]).To(Equal("somevalue"))
 	})
+
+	Context("GetActiveCDI tests", func() {
+		createCDI := func(name string, phase sdkapi.Phase) *cdiv1.CDI {
+			return &cdiv1.CDI{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: name,
+				},
+				Status: cdiv1.CDIStatus{
+					Status: sdkapi.Status{
+						Phase: phase,
+					},
+				},
+			}
+		}
+
+		It("Should return nil if no CDI", func() {
+			client := CreateClient()
+			cdi, err := GetActiveCDI(context.Background(), client)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(cdi).To(BeNil())
+		})
+
+		It("Should return single active", func() {
+			client := CreateClient(
+				createCDI("cdi1", sdkapi.PhaseDeployed),
+			)
+			cdi, err := GetActiveCDI(context.Background(), client)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(cdi).ToNot(BeNil())
+		})
+
+		It("Should return success with single active one error", func() {
+			client := CreateClient(
+				createCDI("cdi1", sdkapi.PhaseDeployed),
+				createCDI("cdi2", sdkapi.PhaseError),
```

Comment:
Do we want to ever allow more than 1 CDI CR?

Reply:
Given that CDI resource is cluster scope, two in "Deployed" state would be very bad

Reply:
I am saying don't allow 2 of any state

Reply:
We can't currently stop CDI resources from being created.  That would require that cdi-operator run a validating webhook.  And even with a webhook it is an inherently racy operation

Reply:
Yeah I mean just return err in this case so we don't ever get further than GetActiveCDI
when that's the case

Reply:
How would that be better?  If CDI A is successfully deployed at t0 and someone creates CDI B at t1 why break reconcile for CDI A?

Reply:
I believe this assumption is made multiple times across the project so if you do allow this,
bad things will happen for CDI A regardless

Reply:
Please explain what bad things would happen and how?

Reply:
I wrote that operator and that function with the intention that there would only be one "active" (deployed phase) CDI and there could be many CDIs in error state

Reply:
oh, I see, even prior to this PR we did not error over this case